src/lib/metaphysics.coffee

```diff
@@ -3,6 +3,19 @@ qs = require 'qs'
 request = require 'superagent'
 { extend, some } = require 'underscore'
 { METAPHYSICS_ENDPOINT, API_REQUEST_TIMEOUT, REQUEST_ID } = require('sharify').data
+ip = require 'ip'
+
+resolveIPv4 = (ipAddress) ->
+  if ip.isV6Format(ipAddress)? and ~ipAddress.indexOf('::ffff')
```

Comment:
The `~` confuses me a little bit :(

In coffeescript it might be more idiomatic to `and ipAdress.indexof('::ffff') is not -1`.

Additionally, these methods can be extracted into a new TypeScript (!) file in Force, and then just required here.

Reply:
I suppose the tilda trick is a bit confusing as it's a bitwise operation - I only used it with `indexOf` to make it truthy - see https://stackoverflow.com/questions/12299665/what-does-a-tilde-do-when-it-precedes-an-expression

Lulz and here I was getting frustrated trying to port the function from Typescript to Coffeescript in the first place - it's not really clear that we can write Typescript here as I didn't see any `.ts` files.  Given the very basic nature of the functions is it ok to leave them functions as-is and port the whole module when we get around to a more general refactor?

Reply:
Yea, in TS or modern JS we could just do `includes`.

I'm cool with waiting on a bigger refactor of this module, for sure!